Bindings/Java/Matlab/Utilities/osimC3Dconverter.m

```diff
@@ -0,0 +1,278 @@
+    markers = tables.get('markers');
+    forces = tables.get('forces');
+
+    %% Rotate marker and force data
+    if isempty(axis)
+        display('Either no axis or value for rotation has been been set; no rotations will be applied')
+        forces_rot = forces.clone();
+        markers_rot = markers.clone();
+    else
+        display(['Rotating Marker and Force data ' num2str(value) ' about the ' axis ' axis']); 
+        markers_rot = osimRotateTableData(markers, axis, value);
+        forces_rot = osimRotateTableData(forces, axis, value);
+    end
+    
+    %% Print the rotated markers to trc file
+    TRCFileAdapter().write(markers_rot,fullfile(path,[filename '.trc']));
+    display([[filename '.trc'] ' written to dir: ' path]);
+    %% Print the force data as a Vec3 sto file and a flattened doubles sto file
+    % make postfix string vector for naming colomns
+
+    %% flatten the Vec3 table to a table of doubles. 
+    postfix = StdVectorString(); postfix.add('_x');postfix.add('_y');postfix.add('_z');
+    forces_flattened = forces_rot.flatten(postfix);
+
+    % update the column names to have v,p,m n them.
+
+    labels = forces_flattened.getColumnLabels();
+    newlabels = forces_flattened.getColumnLabels();
+
+    for i = 0 : labels.size() - 1
+
+        label = char(labels.get(i));
+
+        if i < 9
+            n = '1';
+        elseif  8 < i < 18
+            n = '2';
+        elseif 17 < i < 27
+           n = '3';
+        end
+
+        if ~isempty(strfind(label,'f'))
+            s = ['ground_force_' n '_v'];
+        elseif ~isempty(strfind(label,'p'))
+            s = ['ground_force_' n '_p'];
+        elseif ~isempty(strfind(label,'m'))
+            s = ['ground_torque_' n '_m'];
+        else
+            error(['Column name ' label ' isnt recognized as a force, point, or moment'])
+        end
+            % Get the index for the underscore
+            % in = strfind(label,'_');
+            % add the specifier (f,p,or m) to the label name. 
+            %label_new = [label(1:in) s label(in+1:end)];
+            label_new = [ s label(end)];
+
+            % update the label name 
+            newlabels.set(i,label_new);
+    end
+    % set the column labels
+    forces_flattened().setColumnLabels(newlabels)
+
+    %% Change the points from mm to m
+    for i = 0 : labels.size() - 1
+        label = char(forces_flattened.getColumnLabels().get(i));
+        if ~isempty(strfind(label,'p'))
+            for u = 0 : forces_flattened.getNumRows()-1
+                % Get the point data
+                point = forces_flattened.getDependentColumnAtIndex(i).get(u);
+                % Divide the point by 1000 and set it back into the colomn
+                forces_flattened.getDependentColumnAtIndex(i).set(u,point/1000);
+            end
+        end
+    end
+
+    %% change the order of the forces
+    % turn the timetable into a Maltab Struct
+    sdata = osimTableToStruct(forces_flattened);
```

Comment:
I investigated further and you are right that the force and point must be sequential in the preview code, [GUI #336](https://github.com/opensim-org/opensim-gui/issues/336) which is the source of this problem. I had tested reordering the moments putting one at the beginning and one at the end, so it still preserved the sequence of columns from `_vx` through to `_pz`. Sorry that I had not probed deeper at that time.

Since we cannot go back to 3.3 to fix, I propose changing the order in which the `C3DFileAdaper` sets its columns since force, point, moment is just as legitimate as force, moment, point and would avoid this whole issue of reordering. I will do that.